### DISCUSSION TITLE
Reference tables from other Quarto document

### DISCUSSION THREAD

**Discussion by lorenzoFabbri** [quarto-dev/quarto-cli#7241] [Q&A]
### Description

I am writing a paper in Quarto. My current setup is this: I have one Quarto file (`tables.qmd`) to create all the tables for my manuscript, and another Quarto file (`results.qmd`) for the "Results" section of the manuscript. The file `results.qmd` is then *called* in `manuscript.qmd` (the main Quarto file) with

````
# Results {#sec-res}
```{r}
#| child: output/paper/sections/results.qmd
```
````

If I want to reference a table from `tables.qmd` in `results.qmd`, how do I do that?

**Comment by mcanouil** [comment]
You should use the include shortcode not knitr child as any Quarto content won't be parsed otherwise.

There is a feature in 1.4 to include parts of a Quarto document using the shortcode I mentioned above.
See the documentation.
(Away from a keyboard at the moment)

**Reply by lorenzoFabbri** [reply]
Okay I will remove the child.

Do you mean I can cross reference "objects" from separate files? Because right now the tables.qmd file is not "called" anywhere so obviously it's not working.

**Reply by lorenzoFabbri** [reply]
I am just reading about the embed shortcode.

**Reply by lorenzoFabbri** [reply]
Okay, I think I figured it out. I add `{{< embed tables.qmd >}}` in `manuscript.qmd` and the cross-ref works. What's the difference between `embed` and `include` in this case?

**Reply by cscheid** [reply]
https://quarto.org/docs/authoring/notebook-embed.html

They're two features with different goals and use cases. `embed` embeds the _results_, and `include` includes the entire markdown itself, causing code cells to re-execute, etc.

Embed can reference jupyter notebooks and .qmd files, as well as _parts_ of notebooks and .qmd files.

**Reply by lorenzoFabbri** [reply]
I see, thanks for the references. Is it possible to embed without showing anything? I explain. When submitting a scientific publication, usually the figures should be submitted in a separate file or as individual figures, whereas tables can be "appended" to the main manuscript. So tables do not represent an issue since I can use `embed` after the references. What about figures? I still need to cross-ref them but without *showing* them.

**Reply by cscheid** [reply]
> What about figures? I still need to cross-ref them but without showing them.

That's not currently possible in Quarto.
